src/shims/windows/foreign_items.rs

```diff
@@ -412,7 +422,31 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     _ => this.invalid_handle("SetThreadDescription")?,
                 };
 
-                this.set_thread_name_wide(thread, &name);
+                // FIXME: use non-lossy conversion
+                this.set_thread_name(thread, String::from_utf16_lossy(&name).into_bytes());
+
+                this.write_null(dest)?;
+            }
+            "GetThreadDescription" => {
+                let [handle, name_ptr] =
+                    this.check_shim(abi, Abi::System { unwind: false }, link_name, args)?;
+
+                let handle = this.read_scalar(handle)?;
+                let name_ptr = this.deref_pointer(name_ptr)?; // the pointer where we should store the ptr to the name
+
+                let thread = match Handle::from_scalar(handle, this)? {
+                    Some(Handle::Thread(thread)) => thread,
+                    Some(Handle::Pseudo(PseudoHandle::CurrentThread)) => this.get_active_thread(),
+                    _ => this.invalid_handle("SetThreadDescription")?,
+                };
+                // FIXME what is the right default thread name? Or should we raise an error?
```

Comment:
The right default would be an empty string. Though in std that would be changed to `<unnamed>` anyways.

Reply:
So on actual Windows it succeeds and returns an empty string when no name was that?
That's easy enough to do. Is this documented anywhere?

Yeah I know this is not observable on std.

Reply:
> Is this documented anywhere?

No.